app/Http/Controllers/CommentController.php

```diff
@@ -82,9 +82,74 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        $comment = \App\Comment::find($id);
+        if (($comment->author->id !== \Auth::user()->id && ($comment->post->isLocked() || $comment->flags == 1)) && \Auth::user()->type != 'teacher' && \Auth::user()->type != 'editor'){
+            return redirect()->back()->with('error', 'Error editing comment.: <br>' . $e->getMessage());
+        }
+
+        $comment->content = $request->content;
+        $comment->save();
+
+        return redirect()->back()->with('success', 'Succesfully editing comment!');
+    }
+
+    /**
+     * Update flags form user and comment as flagged
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function flag(Request $request, $id)
+    {
+        $comment = \App\Comment::find($id);
+        if (!$comment->isFlagged())
+        {
+            $comment->increment('flags');
+
+            $user = \Auth::user();
+            $user->Comment_flags()->attach($comment->id, ['reason' => $request->get('content')]);
+        }
+        return back();
+    }
+
+    /**
+     * Update flags form user and comment as flagged
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function change(Request $request, $id)
+    {
+        $comment = \App\Comment::find($id);
+        if ($comment->flags == 1){
+            $comment->increment('flags');
+            $user = Auth::user();
+            $user->Comment_flags()->attach($comment->id, ["reason" => $request->get('content') == null ? "this is a placeholder" : $request->get('content'), "action" => 2, "add_flag_id" => $request->get('flag_id')]);
+        }
+        return back();
+    }
+    /**
+     * Update flags form user and comment as flagged
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function removal(Request $request, $id)
+    {
+        $comment = \App\Comment::find($id);
+        if ($comment->flags == 1 || $comment->flags == 3){
+            $comment->flags = ($comment->flags == 1) ? 3 : 1;
+            $comment->save();
+            $user = Auth::user();
+            $user->Comment_flags()->attach($comment->id, ["reason" => $request->get('content') == null ? "this is a placeholder" : $request->get('content'), "action" => $comment->flags, "add_flag_id" => $request->get('flag_id')]);
```

Comment:
mind the conventions: commentFlags ratherh than Comment_flags. Also is 'content' the reason? Could it be named something like 'reason' instead? Makes more sense in this case.